spring-boot-project/spring-boot/src/main/java/org/springframework/boot/web/servlet/context/ServletWebServerApplicationContext.java

```diff
@@ -170,11 +170,12 @@ protected void onRefresh() {
 	protected void doClose() {
 		if (isActive()) {
 			AvailabilityChangeEvent.publish(this, ReadinessState.REFUSING_TRAFFIC);
-		}
-		super.doClose();
-		WebServer webServer = this.webServer;
-		if (webServer != null) {
-			webServer.destroy();
+			super.doClose();
+
+			WebServer webServer = this.webServer;
+			if (webServer != null) {
+				webServer.destroy();
+			}
```

Comment:
This change will affect every embedded web server, not just Jetty. It either shouldn't be made at all, or if it's really needed, it should be made separately. Please revert it in this PR.